addons/account/wizard/account_payment_register.py

```diff
@@ -149,16 +148,38 @@ def _get_batches(self):
             raise UserError(_("You can't open the register payment wizard without at least one receivable/payable line."))
 
         batches = {}
+
         for line in lines:
             batch_key = self._get_line_batch_key(line)
+            frozendict_key = frozendict(batch_key)
+            batches.setdefault(frozendict_key, {})
+            batches[frozendict_key].setdefault(
+                line.move_id.partner_bank_id.id,
+                {
+                    'key_values': batch_key,
+                    'lines': self.env['account.move.line'],
+                }
+            )
+            batches[frozendict_key][line.move_id.partner_bank_id.id]['lines'] += line
+
+        res = []
+        for partner_bank_ids in batches.values():
+            # batch lines with no partner_bank_id with other lines if they have one same partner_bank_id
+            # this allows to batch a refund (with no partner_bank_id) to bills sharing one same partner_bank_id
```

Comment:
If lines have no partner_bank_id, add them to another batch from the same partner/currency

____

Why mentioning refunds? The field is editable on all documents.